xwiki-platform-core/xwiki-platform-user/xwiki-platform-user-default/src/main/java/org/xwiki/user/internal/document/DocumentDocumentReferenceUserReferenceResolver.java

```diff
@@ -44,18 +45,23 @@
     @Inject
     private EntityReferenceProvider entityReferenceProvider;
 
+    @Inject
+    private EntityReferenceFactory entityReferenceFactory;
+
     @Override
     public UserReference resolve(DocumentReference rawReference, Object... parameters)
     {
         UserReference reference;
         if (rawReference == null) {
             reference = CurrentUserReference.INSTANCE;
         } else {
-            reference = resolveName(rawReference.getName());
+            // small perf improvment to avoid keep duplicated references in memory.
```

Comment:
This change is not needed for this work, it's just that I noticed the dedup of references we were doing in XWikiDocument#setAuthorReference and I thought it should be good to dedup them also in this resolver.